huaweicloud/services/dds/resource_huaweicloud_dds_backup.go

```diff
@@ -0,0 +1,550 @@
+				Type:     schema.TypeString,
+				Computed: true,
+				Description: `Indicates the end time of the backup. The format is yyyy-mm-dd hh:mm:ss.
+The value is in UTC format.`,
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the backup status.`,
+			},
+			"size": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `Indicates the backup size in KB.`,
+			},
+		},
+	}
+}
+
+func BackupDatastoreSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the DB engine.`,
+			},
+			"version": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the database version. The value can be 4.2, 4.0, or 3.4.`,
+			},
+		},
+	}
+	return &sc
+}
+
+func resourceDdsBackupCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// createBackup: create DDS backup
+	var (
+		createBackupHttpUrl = "v3/{project_id}/backups"
+		createBackupProduct = "dds"
+	)
+	createBackupClient, err := cfg.NewServiceClient(createBackupProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DDS Client: %s", err)
+	}
+
+	createBackupPath := createBackupClient.Endpoint + createBackupHttpUrl
+	createBackupPath = strings.ReplaceAll(createBackupPath, "{project_id}", createBackupClient.ProjectID)
+
+	createBackupOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	createBackupOpt.JSONBody = utils.RemoveNil(buildCreateBackupBodyParams(d))
+
+	var createBackupResp *http.Response
+	instanceId := d.Get("instance_id").(string)
+	for {
+		createBackupResp, err = createBackupClient.Request("POST", createBackupPath, &createBackupOpt)
+		if err == nil {
+			break
+		}
+		// if the HTTP response code is 403 and the error code is DBS.201015 or DBS.201014, then it indicates that other
+		// operation is being executed and need to wait
+		if errCode, ok := err.(golangsdk.ErrDefault403); ok {
+			var backupErr backupError
+			err = json.Unmarshal(errCode.Body, &backupErr)
+			if err != nil {
+				return diag.Errorf("error creating DdsBackup: error format error: %s", err)
+			}
+			if backupErr.ErrorCode == "DBS.201014" || backupErr.ErrorCode == "DBS.201015" {
+				err = waitForInstanceRunning(ctx, d, cfg, region, instanceId, schema.TimeoutCreate)
+				if err != nil {
+					return diag.FromErr(err)
+				}
+				continue
+			}
+		}
+		return diag.Errorf("error creating DdsBackup: %s", err)
+	}
+
+	createBackupRespBody, err := utils.FlattenResponse(createBackupResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, err := jmespath.Search("backup_id", createBackupRespBody)
+	if err != nil {
+		return diag.Errorf("error creating DdsBackup: ID is not found in API response")
```

Comment:
```suggestion
		return diag.Errorf("error creating DDS backup: ID is not found in API response")
```

please update it in all files

Reply:
done